### PR TITLE
chore: Update CODEOWNERS file to enforce AION SDK team reviews on shared files.

### PR DESCRIPTION
**Pull request by clundin25**

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -4,10 +4,27 @@
...
+tests/test_aws.py                                           @googleapis/googleapis-auth @googleapis/aion-sdk
+tests/test_credentials.py                                   @googleapis/googleapis-auth @googleapis/aion-sdk
+tests/test_downscoped.py                                    @googleapis/googleapis-auth @googleapis/aion-sdk
+tests/test_external_account.py                              @googleapis/googleapis-auth @googleapis/aion-sdk
+tests/test_external_account_authorized_user.py              @googleapis/googleapis-auth @googleapis/aion-sdk
+tests/test_identity_pool.py                                 @googleapis/googleapis-auth @googleapis/aion-sdk
+tests/test_pluggable.py                                     @googleapis/googleapis-auth @googleapis/aion-sdk
+tests/test_sts.py                                           @googleapis/googleapis-auth @googleapis/aion-sdk
+tests/test_impersonated_credentials.py                      @googleapis/googleapis-auth @googleapis/aion-sdk
+/samples/                                                   @googleapis/python-samples-owners
```
Comment (dandhlee) [inline .github/CODEOWNERS:29]:
```suggestion
/samples/                                                   @googleapis/python-samples-reviewers 
```

Comment (clundin25) [reply .github/CODEOWNERS:29]: What's the context behind this change @dandhlee?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -4,10 +4,27 @@
 # For syntax help see:
 # https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax
 
-# The @googleapis/yoshi-python is the default owner for changes in this repo
-*               @arithmetic1728 @sai-sunder-s @googleapis/googleapis-auth @googleapis/yoshi-python
+# The @googleapis/googleapis-auth and @googleapis/yoshi-python is the default owner for changes in this repo
+*                                                           @googleapis/googleapis-auth @googleapis/yoshi-python
+google/auth/_default.py                                     @googleapis/googleapis-auth @googleapis/aion-sdk
```
Comment (sai-sunder-s) [inline .github/CODEOWNERS:9]: There is no way to make both reviewers required for _default.py right?

Comment (clundin25) [reply .github/CODEOWNERS:9]: Doesn't seem like that is supported https://github.com/isaacs/github/issues/1205

**Review by sai-sunder-s — APPROVED** [review]

**Pull request merged**